lib/task_helper.rb

```diff
@@ -0,0 +1,48 @@
+require 'json'
+
+class TaskHelper
+  class Error < RuntimeError
+    attr_reader :kind, :details, :issue_code
+
+    def initialize(msg, kind, details = nil)
+      super(msg)
+      @kind = kind
+      @issue_code = issue_code
+      @details = details || {}
+    end
+
+    def to_h
+      { 'kind' =>  kind,
+        'msg' => message,
+        'details' => details }
+    end
+  end
+
+  def task(params = {})
+    msg = 'The task author must implement the `task` method in the task'
+    error = TaskHelper::Error.new(msg, 'tasklib/not-implemented')
+    STDOUT.print error.to_h.to_json
+    raise error
+  end
+
+  def self.run
+    params = JSON.parse(STDIN.read)
```

Comment:
We should symbolize these keys. In addition to being less surprising, it also allows the `task` method to take keyword arguments.

For instance, an updated version of the package task might look like:

```rb
def task(name:, action:, provider: nil, version: nil)
  # [...]
end
```

That saves a step of having to pull all the parameters out of the hash. It also provides one more bit of assurance that the required keys are in fact present (this is guaranteed by the task spec, but it still feels wrong sometimes to boldly assume it).

Reply:
Also we need to check whether `STDIN.read` actually contained input. If it didn't, we should return an error instructing the user to enable the `stdin` input_method.

Reply:
Will we receive any stdin if there are no parameters?

Reply:
Yeah, if there are no parameters then stdin will have `{}`. Although we should always be setting `_task` and `_installdir` now, so there will even be keys!